superduperdb/components/plugin.py

```diff
@@ -0,0 +1,114 @@
+import importlib.util
+import os
+import shutil
+import subprocess
+import sys
+import typing as t
+
+from superduperdb import Component, logging
+from superduperdb.components.datatype import LazyFile, file_lazy
+
+
+class Plugin(Component):
+    """Plugin component allows to install and use external python packages as plugins.
+
+    :param path: Path to the plugin package or module.
+    :param identifier: Unique identifier for the plugin.
+    :param cache_path: Path to the cache directory where the plugin will be stored.
+    """
+
+    type_id: t.ClassVar[str] = "plugin"
+    _artifacts: t.ClassVar = (("path", file_lazy),)
+    path: str
+    identifier: str = ""
+    cache_path: str = ".superduperdb/plugins"
+
+    def __post_init__(self, db, artifacts):
+        if isinstance(self.path, LazyFile):
+            self._prepare_plugin()
+        else:
+            path_name = os.path.basename(self.path.rstrip("/"))
+            self.identifier = self.identifier or f"plugin-{path_name}"
+        self._install()
+        super().__post_init__(db, artifacts)
+
+    def _install(self):
+        logging.debug(f"Installing plugin {self.identifier}")
+        package_path = self.path
+        module_name = os.path.basename(self.path.rstrip("/"))
+
+        # Check if plugin is already installed
+        check_tag = f"_PLUGIN_{self.uuid}"
+
+        if check_tag in os.environ:
+            logging.debug(f"Plugin {self.identifier} already installed")
+            return
+
+        if os.path.isdir(package_path):
+            import_package_path = os.path.join(package_path, "__init__.py")
+
+            if not os.path.exists(import_package_path):
+                logging.info(f"Creating __init__.py file in {package_path}")
+                open(import_package_path, "w").close()
+
+            logging.debug(f"Plugin {self.identifier} is a package")
+            self._pip_install(package_path)
+        else:
+            module_name = module_name.split(".")[0]
```

Comment:
Should this also support a `requirements.txt` file?

Reply:
Currently unsupported, do we need this feature? It seems like you want to only install the requirements without any Python modules.

Reply:
Supported now

```python
from superduperdb.components.plugin import Plugin
# All files with the requirements.txt suffix must be specified.
plugin = Plugin(path="deploy/installations/testenv_requirements.txt")
```